app/models/accounting/transaction.rb

```diff
@@ -208,6 +207,14 @@ def net_debit_for_account(account_id)
     end
   end
 
+  def disbursement_fields
+    if loan_transaction_type_value == "disbursement"
+      errors.add(:disbursement_type) unless disbursement_type.present?
+      errors.add(:qb_vendor_id) unless qb_vendor_id.present?
+      errors.add(:qb_object_type, "must be purchase") unless qb_object_type == "Purchase"
```

Comment:
Typically if a value is forced to be something, we shouldn't really be doing a validation error, but instead:

1. Either hiding or disabling the form field and
2. Forcing the value to be "Purchase" in a before_validation callback. I usually call this a 'normalization' step.

If the client-side JS part of that feels like too much right now, we could just add the normalization bit (cover it with a quick spec that tests both disbursement and non-disbursement flows (I can share some nice idiomatic code for this) and just do it without telling the user. I think that is still better than adding a new validation, translating it, testing it, etc. At least we're part way there..? Curious for your thoughts...

Reply:
Oh, hold on, it seems `qb_object_type` is not in the form... Let's talk about this one...